src/editor/input_center.cpp

```diff
@@ -0,0 +1,738 @@
+//  SuperTux
+//  Copyright (C) 2015 Hume2 <teratux.mail@gmail.com>
+//
+//  This program is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+#include "editor/input_center.hpp"
+
+#include "control/input_manager.hpp"
+#include "editor/editor.hpp"
+#include "editor/object_menu.hpp"
+#include "editor/point_marker.hpp"
+#include "editor/tool_icon.hpp"
+#include "editor/tip.hpp"
+#include "math/rectf.hpp"
+#include "object/ambient_sound.hpp"
+#include "object/camera.hpp"
+#include "object/path.hpp"
+#include "object/platform.hpp"
+#include "object/coin.hpp"
+#include "object/tilemap.hpp"
+#include "object/wind.hpp"
+#include "gui/menu.hpp"
+#include "gui/menu_manager.hpp"
+#include "supertux/menu/menu_storage.hpp"
+#include "supertux/game_object.hpp"
+#include "supertux/game_object_ptr.hpp"
+#include "supertux/level.hpp"
+#include "supertux/moving_object.hpp"
+#include "supertux/object_factory.hpp"
+#include "supertux/sector.hpp"
+#include "supertux/tile.hpp"
+#include "supertux/tile_manager.hpp"
+#include "supertux/tile_set.hpp"
+#include "trigger/climbable.hpp"
+#include "trigger/scripttrigger.hpp"
+#include "trigger/secretarea_trigger.hpp"
+#include "trigger/sequence_trigger.hpp"
+#include "video/font.hpp"
+#include "video/renderer.hpp"
+#include "video/video_system.hpp"
+#include "math/vector.hpp"
+
+EditorInputCenter::EditorInputCenter() :
+  hovered_tile(0, 0),
+  sector_pos(0, 0),
+  mouse_pos(0, 0),
+  dragging(false),
+  drag_start(0, 0),
+  dragged_object(NULL),
+  marked_object(NULL),
+  edited_path(NULL),
+  object_tip(),
+  obj_mouse_desync(0, 0),
+  render_grid(true),
+  scrolling(Vector())
+{
+}
+
+EditorInputCenter::~EditorInputCenter()
+{
+}
+
+void
+EditorInputCenter::update(float elapsed_time) {
+  update_scroll();
+}
+
+void
+EditorInputCenter::delete_markers() {
+  for (auto i = Editor::current()->currentsector->moving_objects.begin();
```

Comment:
Please:
auto sector = Editor::current()->currentsector; 

and then replace the occurrences in the for loop.